docs/en/engines/database-engines/materialize-mysql.md

```diff
@@ -0,0 +1,160 @@
+| VARCHAR, VAR_STRING     | [String](../../sql-reference/data-types/string.md)           |
+| BLOB                    | [String](../../sql-reference/data-types/string.md)           |
+
+Other types are not supported. If MySQL table contains a column of such type, ClickHouse throws exception "Unhandled data type" and stops replication.
+
+[Nullable](../../sql-reference/data-types/nullable.md) is supported.
+
+## Specifics and Recommendations {#specifics-and-recommendations}
+
+### DDL Queries
+
+MySQL DDL queries are converted into the corresponding ClickHouse DDL queries ([ALTER](../../sql-reference/statements/alter/index.md), [CREATE](../../sql-reference/statements/create/index.md), [DROP](../../sql-reference/statements/drop.md), [RENAME](../../sql-reference/statements/rename.md)). If ClickHouse cannot parse some DDL query, the query is ignored.
+
+### DML Queries
+
+`MaterializeMySQL` engine supports only `INSERT` and `SELECT` queries.
+
+MySql `DELETE` query is converted into `INSERT` with `_sign=-1`. 
+
+MySQL `UPDATE` query is converted into `INSERT` with `_sign=-1` and `INSERT` with `_sign=1`.
+
+`SELECT` query has some specifics:
+
+- If `_version` is not specified in the `SELECT` query, [FINAL](../../sql-reference/statements/select/from.md#select-from-final) modifier is used. So only rows with `MAX(_version)` are selected.
+
+- If `_sign` is not specified in the `SELECT` query, `WHERE _sign=1` is used by default, so the deleted rows are not included into the result set. 
+
+### Index Conversion
+
+MySQL `PRIMARY KEY` and `INDEX` clauses are converted into `ORDER BY` tuples in ClickHouse tables.
+
+ClickHouse has only one physical order, which is determined by `ORDER BY` clause. To create a new physical order, use [materialized views](../../sql-reference/statements/create/view.md#materialized).
+
+ **Notes**
+
+ - Rows with `_sign=-1` are not deleted physically from the tables. 
+ - Cascade `UPDATE/DELETE` queries are not supported by the `MaterializeMySQL` engine.
+ - Replication can be easily broken.
+ - Manual operations on database and tables are forbidden.
+
+## Examples of Use {#examples-of-use}
+
+Table in MySQL:
+
+``` text
+mysql> CREATE DATABASE db;
+mysql> CREATE TABLE db.test (a INT PRIMARY KEY, b INT);
+mysql> INSERT INTO db.test VALUES (1, 11), (2, 22);
+mysql> DELETE FROM db.test WHERE a=1;
+mysql> ALTER TABLE db.test ADD COLUMN c VARCHAR(16);
+mysql> UPDATE db.test SET c='Wow!', b=222;
+mysql> SELECT * FROM test;
+
++---+------+------+ 
+| a |    b |    c |
++---+------+------+ 
+| 2 |  222 | Wow! |
++---+------+------+
+```
+
+Database in ClickHouse, exchanging data with the MySQL server:
+
+``` sql
+CREATE DATABASE mysql ENGINE = MaterializeMySQL('localhost:3306', 'db', 'user', '***');
+SHOW TABLES FROM mysql;
+```
+
+``` text
+┌─name─┐
+│ test │
+└──────┘
+```
+
+``` sql
+SELECT * FROM mysql.test;
+```
+
+``` text
+┌─a─┬──b─┐ 
+│ 1 │ 11 │ 
+│ 2 │ 22 │ 
+└───┴────┘
+```
+
+``` sql
+SELECT * FROM mysql.test;
```

Comment:
The query is the same as the previous one, but results are different. We should explain why.
(because we've got the first result just after `mysql> INSERT INTO db.test VALUES (1, 11), (2, 22);` and the second one after `mysql> UPDATE db.test SET c='Wow!', b=222;`)